### DISCUSSION TITLE
A few changes? (React, Electron, etc)

### DISCUSSION THREAD

**Discussion by garrettcharliehill** [karlcswanson/micboard#31] [Ideas]
Hey @karlcswanson!

A friend of mine loves using your software and we use it in production at our church. We recently forked the code and made some local changes (a small formatting change and drag-and-drop image change support).
- It's really basic but [you can check it out here](https://github.com/garrettcharliehill/micboard).

I'm curious if you're interested in working on any of the following changes:
- Separating the server code from the UI code 
- Transitioning the front end UI to a modern javascript framework like React
- Improve standalone functionality (installer for linux and windows - no browser)

I'd be more than happy to help contribute. I love what you've built here and would be happy to help out as much or as little as you want.

Thanks for building this!
Charlie Hill

**Comment by karlcswanson** [comment]
Hello @garrettcharliehill!
I have thought about all of those improvements.  The main hold up is test hardware.  When I made Micboard, I had easy access to QLX-D, Axient Digital, PSM 1000, and ULX-D through work and church.  I've since moved a few states away.  I'm hesitant to push changes without running through basic tests.  Coincidentally, I work at a large touring audio provider now.  There may be some possibilities there.

You may be interested in the [0.8.7-updates](https://github.com/karlcswanson/micboard/tree/0.8.7-updates) branch.  Most of this just involves bringing dependencies up to date.  I also removed electron and added in multistage automated docker builds with GitHub Actions.  Going forward, I think distribution will be via docker.  Packaging the electron app was a complicated manual process.

Here is a [docker-compose](https://gist.github.com/karlcswanson/649b7cd8b38ef904a3a5ad433565939a) if you are interested in trying the new branch via docker.

Drag and drop looks great! That has been on my list for a bit.  Once I get the current branch merged in, maybe i can get that added too.

**Reply by TreyCarey** [reply]
Is there an ETA for when you plan on merging it?

**Reply by karlcswanson** [reply]
No, but I have been tinkering with a Shure device emulator the past few weeks,  I have it emulating 4 Axient Digital units.  The plan is to use this to develop micboard, then test against a rack of RF devices when I have access to that type of equipment.

**Reply by TreyCarey** [reply]
Great to hear from you again, if you ever need I have Dual Axient, Quad Axient, and Quad ULXD if you need someone to test things out

**Reply by s0600204** [reply]
> [...] tinkering with a Shure device emulator [...]

@karlcswanson, May I ask if this emulator is available anywhere? (I've been investigating adding Sennheiser support, and I need some way of checking my changes aren't breaking compatibility for Shure devices!)

**Reply by karlcswanson** [reply]
Not available anywhere, I got kinda far with it but I didn't have any hardware to validate it against.  It was a golang program that took a few audio files as inputs and sent out meter strings like the receivers.

**Reply by s0600204** [reply]
Ah. Well, thought it worth asking. Thanks for the info.
